homeassistant/components/esphome/__init__.py

```diff
@@ -652,6 +653,7 @@ def from_hass(self, value: _ValT) -> _EnumT:
         EntityCategory.NONE: None,
         EntityCategory.CONFIG: ENTITY_CATEGORY_CONFIG,
         EntityCategory.DIAGNOSTIC: ENTITY_CATEGORY_DIAGNOSTIC,
+        EntityCategory.SYSTEM: ENTITY_CATEGORY_SYSTEM,
```

Comment:
`system` should not be used, it is for internal use inside Home Assistant.

Reply:
Thanks @frenck

Reply:
Is the `system` entity category meant to only be used internally by Home Assistant? If so, it'd be nice if that was explicitly mentioned in the docs. As it stands, it says "Set to `system` for an entity which is not useful for the user to interact with", which as I interpret it could also apply to integration-defined entities. E.g. one use case I had in mind was that I've some entities in ESPHome that are used to track state in some complex automations. Right now I've marked them as `internal` in ESPHome so that they're not exposed to HA, but this would allow me to expose them without them getting in the way, with the added benefit that I can access them from HA automations as well, and that when I'm debugging or something broke I can view/change state directly in HA.

There's already an [integration](https://github.com/home-assistant/core/blob/6cba03aa4a45b49420b77ae405f52138b8a499df/homeassistant/components/nut/const.py#L229) providing system entities, btw.

Reply:
> Is the system entity category meant to only be used internally by Home Assistant? 

Yes/no. It hides the entity from almost all places. It is used by, for example, the internal entities that track the costs of the energy dashboard. It is meant for entities that are not for user interaction (as the docs state).

Adding such entities from ESPHome will therefore not match IMHO and only cause confusion.

Reply:
> Yes/no. It hides the entity from almost all places. It is used by, for example, the internal entities that track the costs of the energy dashboard. It is meant for entities that are not for user interaction (as the docs state).

Well, that's a bit vague criteria, but I can think of a few usecases where that's exactly what you'd want (e.g. the example I gave). If they aren't meant to only be used by HA itself internally and integrations are allowed to provide them, I wouldn't exclude ESPHome.

I agree it doesn't make sense if you just use ESPHome as a gateway for sensors/switches in HA, but if you're using ESPHome to build "smart" devices that have complex state and logic, I think it does make sense to allow making system entities. As for confusion, we've plenty of worse footguns already, and the option is hidden behind a link in a docs, so I don't expect people that aren't explicitly looking for it will try to use this.

Reply:
Opened up a PR to remove it from NUT, which is incorrect indeed.

Reply:
@oxan There is really no use in adding them to ESPHome. The only thing it will cause is confusion.